shell/platform/darwin/macos/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -403,9 +456,19 @@ - (NSAttributedString*)attributedSubstringForProposedRange:(NSRange)range
 }
 
 - (NSRect)firstRectForCharacterRange:(NSRange)range actualRange:(NSRangePointer)actualRange {
-  // TODO: Implement.
-  // Note: This function can't easily be implemented under the system-message architecture.
-  return CGRectZero;
+  // This only determines position of caret instead of any arbitrary range, but it's enough
+  // to properly position accent selection popup
+  if (CATransform3DIsAffine(_editableTransform) && !CGRectEqualToRect(_caretRect, CGRectNull)) {
+    CGRect rect =
+        CGRectApplyAffineTransform(_caretRect, CATransform3DGetAffineTransform(_editableTransform));
+
+    // flip and convert to screen coordinates
+    double viewHeight = self.flutterViewController.view.bounds.size.height;
```

Comment:
self.flutterViewController will be null here ref https://github.com/flutter/engine/pull/23469/files#r612811988

cc @knopp @cbracken

Reply:
I noticed that :)

https://github.com/flutter/engine/pull/25524